{"version":3,"file":"kafka-retriable-exception.js","sourceRoot":"","sources":["kafka-retriable-exception.ts"],"names":[],"mappings":";;;AAAA,mDAA+C;AAE/C;;;;;;;;;GASG;AACH,MAAa,uBAAwB,SAAQ,4BAAY;IAChD,QAAQ;QACb,OAAO,IAAI,CAAC;IACd,CAAC;CACF;AAJD,0DAIC","sourcesContent":["import { RpcException } from './rpc-exception';\r\n\r\n/**\r\n * Exception that instructs Kafka driver to instead of introspecting\r\n * error processing flow and sending serialized error message to the consumer,\r\n * force bubble it up to the \"eachMessage\" callback of the underlying \"kafkajs\" package\r\n * (even if interceptors are applied, or an observable stream is returned from the message handler).\r\n *\r\n * A transient exception that if retried may succeed.\r\n *\r\n * @publicApi\r\n */\r\nexport class KafkaRetriableException extends RpcException {\r\n  public getError(): string | object {\r\n    return this;\r\n  }\r\n}\r\n"]}